### PR TITLE
Ignore import errors if in guarded import block

### LINKED ISSUE #3285: Add option to ignore code in ``if TYPE_CHECKING``
### Is your feature request related to a problem? Please describe
In typed code, this is common pattern:
```python
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from some_module import some_type
```
If `some_type` is defined in stub, it doesn't have to exist in runtime (e. g. `wsgiref.types` is only available in `.pyi` files). Pylint issues an error for this code, while it's perfectly valid.
Pylint isn't a type checker, so it should ignore all code in `if TYPE_CHECKING:`.

### Describe the solution you'd like
Add option like to ignore all errors or, probably, only import-related errors in `if TYPE_CHECKING:` blocks.

### LINKED ISSUE #3382: Pylint reports ungrouped-imports for TYPE_CHECKING imports
### Steps to reproduce
```py
from typing import TYPE_CHECKING
from anyio import run, connect_tcp, create_task_group

if TYPE_CHECKING:
    from typing import List
    from anyio import SocketStream, TaskGroup  # pylint: disable=ungrouped-imports
```
Remove the pylint: disable comment and it will report ungrouped-imports

### Current behavior

ungrouped-imports is reported for any duplicate import sources defined in an `if TYPE_CHECKING` block, not including the `typing` module.

### Expected behavior

Pylint accepts the syntax as it's common to have certain imports in TYPE_CHECKING if not needed at runtime.

### pylint --version output

```
pylint 2.4.4
astroid 2.3.3
Python 3.8.1 (default, Jan  8 2020, 23:09:20) 
[GCC 9.2.0]
```

### PR DESCRIPTION
**Pull request by cdce8p**
## Description
Ignore the following error messages if import is in guarded if block (`sys.version_info` or `typing.TYPE_CHECKING`):
* `import-error`
* `no-name-in-module`
* `ungrouped-imports`

Although just ignoring them isn't ideal, that seems to be the best we can do here. Besides, a fix seems to be highly requested.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Related Issue
Closes #3285
Closes #3382

### CODE REVIEW

**Comment by cdce8p** [comment]
@Pierre-Sassoulas What do you think? I'm open to other ideas if you have any

File: pylint/checkers/utils.py [hunk 1]
```diff
@@ -1535,3 +1535,28 @@ def get_import_name(
                     modname, level=importnode.level
                 )
     return modname
+
+
+def is_node_in_guarded_import_block(node: astroid.NodeNG) -> bool:
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/utils.py:1540]: Would it make sense to have this check directly in ``astroid.If`` as ``is_typing_or_sys_guard`` ? Would permit to refactor to:
```python
return isinstance(node.parent, astroid.If) and node.is_typing_or_sys_guard()
```
Or even use directly without creating a function.

**Review by Pierre-Sassoulas — APPROVED** [review]
Good job on adding the typing, little by little we'll end up having proper typing :D

File: pylint/checkers/utils.py (hunk 1 above)
Comment (cdce8p) [reply pylint/checkers/utils.py:1540]: See PyCQA/astroid#1099

**Pull request merged**